packages/ast-spec/src/declaration/VariableDeclaration/spec.ts

```diff
@@ -29,5 +29,5 @@ export interface VariableDeclaration extends BaseNode {
    * var z = 3;
    * ```
    */
-  kind: 'const' | 'let' | 'var';
+  kind: 'await using' | 'const' | 'let' | 'using' | 'var';
```

Comment:
In theory, we could adhere more strictly to the [Explicit Resource Management > Declarations > VariableDeclaration spec](https://github.com/estree/estree/blob/0362bbd130e926fed6293f04da57347a8b1e2325/stage3/explicit-resource-management.md#variabledeclaration):

> If `kind` is `"using"` or `"await using"`, for every declarator `d` of `declarations`, `d.id` must be an Identifier. If the variable declaration is the `left` of a ForOfStatement, `d.init` must be `null`, otherwise `d.init` must be an Expression.

I'm not familiar with our practices around specifics like that. Do we normally?

Without any prior context, I would in theory prefer to make our AST more strict & compliant (at the cost of having more types). That way AST types are more accurate. For example, if someone is writng a rule around `using` statements, they probably wouldn't want to have to assert `declarations.id` to be `Identifier` instead of the default.

Reply:
yup I'd definitely agree here!
the spec is much stricter, so we can separate the types to make things easier to refine.
we can also enforce this from the parsing side as well - eg error if the `id` is not an `Identifier`.

Reply:
Looks like the latest commit gets the first bit around `.id`s, but not the ForOfStatement narrowing. Progress!